UPGRADE.md

```diff
@@ -2,6 +2,25 @@
 
 ## dev-master
 
+### Add key property to Role entity
+
+To allow for referencing `Role` entities by a human readable string, a `key` property was added to the `Role` entity.
+Furthermore, the `key` property is used for generating the identifier of a role instead of the `name` property now.
+The following statements update the database to include the new property and set the key of each existing Role to 
+keep backwards compatibility:
+
+```sql
+DROP INDEX UNIQ_13B749A05E237E06 ON se_roles;
+ALTER TABLE se_roles ADD `role_key` VARCHAR(60) DEFAULT NULL;
+UPDATE se_roles SET `role_key` = `name` WHERE `role_key` IS NULL;
```

Comment:
this would not be needed when making it nullable

Reply:
it is not needed, but by doing this i dont need to fallback to the `name` in the `getIdentifier` method. but i am okay with implementing a fallback too

Reply:
I would definitely remove this and do a fallback else it could end strange for this who implemented synchronisation scripts.

Reply:
alright 👍